soroban-env-host/src/host.rs

```diff
@@ -1074,6 +1074,31 @@ impl VmCallerCheckedEnv for Host {
         Ok(RawVal::from_void())
     }
 
+    fn log_fmt_values(
+        &self,
+        _vmcaller: &mut VmCaller<Host>,
+        fmt: Object,
+        args: Object,
+    ) -> Result<RawVal, HostError> {
+        if cfg!(debug_assertions) {
+            let fmt: String = self
+                .visit_obj(fmt, move |hv: &Vec<u8>| Ok(String::from_utf8(hv.clone())))?
+                // TODO: Remove unwrap.
+                .unwrap();
```

Comment:
This allocates a String, but only when debug-assertions are enabled. This code will not be compiled in releases used in stellar-core.